### PR TITLE
Make webpack output build errors and listen to it

### PR DESCRIPTION
**Pull request by cjb**
@keybase/react-hackers

Wow, so many ignored errors!

webpack does not return an error or a non-zero exit code on build errors unless you tell it to with:

* `--bail` on the command line (for our `npm run build-prod`), or
* `bail: true` in the webpack config (for our package.js script)

The reasoning supposedly has something to do with using webpack in watch mode but I find that veeeery unconvincing.

Once webpack is returning errors in those two places, we were then ignoring **those** errors in two different places -- the execSync() call in the `npm run build-prod` case, and the (err) argument in the package.js case.

We were doing:
```js
     if (err) {
       return console.error(err)
```
but console.*() always returns undefined, so the return statement doesn't have any effect.

With all four of these changes (two fixes to webpack call, two fixes to checking the result of the webpack call), on Linux, `npm run package` ends at:
```
npm ERR! Failed at the keybase@1.0.1 package script 'babel-node --presets es2015,stage-2 npm-helper.js package'.
```

@gabriel and @oconnor663, please could you check this PR causes your builds to error out and fail?

### CODE REVIEW

**Comment by oconnor663** [comment]
Hmm, when I tried it just now, `packaging/linux/build_binaries.sh prerelease` on your branch succeeded. I cleaned out all my node modules and tried again, but it succeeded again.

**Comment by cjb** [comment]
Ah, you also need to revert the JS code fix from Gabriel on Sat..

**Comment by cjb** [comment]
`git revert a203551485730`

didn't push the revert for obvious broken-build reasons :)

**Comment by oconnor663** [comment]
Success! Which is to say...failure!

```
{ [ModuleBuildError: Module build failed: TypeError: /home/jacko/go/src/github.com/keybase/client/
shared/reducers/tracker.js: Duplicate declaration "username"
  296 |           return state
  297 |         }
> 298 |         const username = action.payload.username
      |               ^
  299 | 
  300 |         return {
  301 |           ...state,
    at File.buildCodeFrameError (/home/jacko/go/src/github.com/keybase/client/desktop/node_modules
/babel-core/lib/transformation/file/index.js:407:15)
...
```

File: desktop/npm-helper.js [hunk 1]
```diff
@@ -113,7 +113,7 @@ const commands = {
   'build-prod': {
     nodeEnv: 'production',
     nodePathDesktop: true,
-    shell: 'webpack --config webpack.config.production.js --progress --profile --colors',
+    shell: 'webpack --config webpack.config.production.js --progress --profile --colors --bail',
```
Comment (chrisnojima) [inline desktop/npm-helper.js]: i'd rather put this into the config

**Comment by cjb** [comment]
@chrisnojima Thanks, you're right, `bail: true` can live in the config only.  Updated the PR.

**Comment by chrisnojima** [comment]
this is fine to me, i want it tested on the mac mini first

**Comment by cjb** [comment]
Cool, same.  Will wait for @gabriel before merging.

**Comment by gabriel** [comment]
Testing now

**Comment by gabriel** [comment]
Yep fails on OS X..  https://gist.github.com/gabriel/2c7c6b9939799ae7c126

**Comment by cjb** [comment]
Thanks!  I'll merge in now, then, yell if anyone would like to wait for anything first.

**Pull request merged**